### DISCUSSION TITLE
Setting certain argtypes as required

### DISCUSSION THREAD

**Discussion by TimGeerts** [storybookjs/storybook#26608] [Help] [answered]
### Summary

 I must have overlooked something really simple, but, how can I set specific argtypes as required so it shows the red asterisk next to the name in the controls?
I'm not autogenerating anything for now, just making my own argtypes and args and combining them in stories.  
Many screenshots I've seen have the "red asterisk" next to specific args in the control table, but I can't seem to figure out how...

**Comment by Marklb** [comment, marked as answer]
I would need to look through the code some more to see if it mentions it anywhere, because I see `argTypes.myArg.type.required` and `argTypes.myArg.table.required`. I think the one set on table is the old way that may need to be updated in the code that sets it there, but I am not positive.

The following seems to work though:

```ts
export default {
  ...,
  argTypes: {
    myArg: {
      type: {
        name: 'string',
        required: true,
      },
    },
  },
}
```

**Reply by daformat** [reply]
Thanks! What if we want to keep the type name inference? It's feels weird to have to redeclare every type when it's actually properly inferred by SB in the first place. Plus some type like `object` require a `value` to be set
